### PR TITLE
ADD: tfsec package

### PR DESCRIPTION
**Pull request by thongngo3301**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [x] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: tfsec.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package:
+  name: tfsec
+  version: 1.28.5
+  epoch: 2
```
Comment (tuananh) [inline tfsec.yaml:4]: new package should have epoch 0

File: tfsec.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        -ldflags "-X github.com/aquasecurity/tfsec/version.Version=${TAG} \
+        -s -w -extldflags '-fno-PIC -static'" ./cmd/tfsec/
+      GOOS=linux GOARCH=amd64 go build \
+        -o bin/linux/${CHECK_GEN_BINARY}-linux-amd64 \
+        -ldflags "-X github.com/aquasecurity/tfsec/version.Version=${TAG} \
+        -s -w -extldflags '-fno-PIC -static'" ./cmd/tfsec-checkgen/
+      mkdir -p ${{targets.destdir}}/usr/bin
+      chmod +x bin/linux/*
+      mv bin/linux/${BINARY}-linux-amd64 ${{targets.destdir}}/usr/bin/${BINARY}
+      mv bin/linux/${CHECK_GEN_BINARY}-linux-amd64 ${{targets.destdir}}/usr/bin/${CHECK_GEN_BINARY}
```
Comment (tuananh) [inline tfsec.yaml:47]: dont you want to rename this to tfsec?

also, this script only compile for amd64 and we have amd64 & arm64 here.

File: tfsec.yaml (hunk 1 above)
Comment (thongngo3301) [reply tfsec.yaml]: updated in new commit

File: tfsec.yaml (hunk 2 above)
Comment (thongngo3301) [reply tfsec.yaml]: i had mv command to rename binary package to tfsec in /usr/bin path
updated multi arch in new commit

File: tfsec.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: 2a2a67987b477eb4a76ff42858973096138e9943
+      repository: https://github.com/aquasecurity/tfsec
+      tag: v${{package.version}}
+
+  - uses: go/bump
+    with:
+      deps: google.golang.org/protobuf@v1.33.0
```
Comment (tuananh) [inline tfsec.yaml:27]: do we need to bump `github.com/golang/protobuf` to v1.5.3 as well?

File: tfsec.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  - runs: |
+      export BINARY=tfsec
+      export CHECK_GEN_BINARY=tfsec-checkgen
+      export TAG=v${{package.version}}
+      export GO111MODULE=on
+      export CGO_ENABLED=0
+      export GOOS=$(go env GOOS)
+      export GOARCH=$(go env GOARCH)
+      mkdir -p bin/${GOOS}
```
Comment (tuananh) [inline tfsec.yaml:37]: no need to makedir i think. go build does that for you.

File: tfsec.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      go build \
```
Comment (tuananh) [inline tfsec.yaml:38]: try to use go/build where possible. this way, if we want to make change to build flag later, it'll be easier

https://github.com/chainguard-dev/melange/blob/main/pkg/build/pipelines/go/build.yaml

Comment (tuananh) [reply tfsec.yaml:38]: go/build also has a flag for bumping dep

File: tfsec.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      ldflags: -X github.com/aquasecurity/tfsec/version.Version=v${{package.version}} -s -w -extldflags '-fno-PIC -static'
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: aquasecurity/tfsec
+    use-tag: true
+    strip-prefix: v
```
Comment (ajayk) [inline tfsec.yaml:48]: can you please add a basic version test 
```suggestion
    strip-prefix: v
 
 test:
  pipeline:
    - runs: |
        tfsec --version | grep ${{package.version}}
```

**Review by ajayk — APPROVED** [review]

**Pull request merged**